drafts/two-way-binding-signals-and-query-params.md

```diff
@@ -0,0 +1,631 @@
+
+Let's take a closer look at this `anonymise` method:
+
+```ts
+const queryParams = { firstName: undefined, lastName: undefined };
+this.router.navigate([], { queryParams, queryParamsHandling: "merge" });
+```
+
+We want to modify the value of a certain query param through Angular's router API. This may surprise you but the easiest way is by triggering an "empty navigation" (notice the `[]`) and specifying the new query params in the configuration object.
+
+Finally, the UI gets updated successfully because, once the router finishes the navigation, a new value will be emitted in the `activatedRoute.queryParams` observable, which we map and convert to the signals that we have in our component. So every time it emits a new value, our signals will be in sync.
+
+This is very cool but we can do better:
+
+```ts
+anonymise() {
+  // It's not very nice that we have to specify the query params again here ⬇️
+  const queryParams = { firstName: undefined, lastName: undefined };
+  this.router.navigate([], { queryParams, queryParamsHandling: "merge" });
+}
+```
+
+Notice that we already have the `firstName` and `lastName` signals in our component. Wouldn't it be nicer if we could just **set** the new values of the query params on the signals directly? Something like this:
+
+```ts
+anonymise() {
+  this.firstName.set(undefined)
+  this.lastName.set(undefined)
+}
+```
+
+That would be amazing, but unfortunately this doesn't work since the `toSignal` utility returns a read-only signal.
+
+That doesn't mean we cannot do it ourselves though. Let's build it!
+
+## Implementing Two-way binding between Signals and Query Params
+
+First let's create a handy `QueryParamsService` so we separate the query param handling logic from our component:
+
+```ts
+@Injectable({ providedIn: "root" })
+export class QueryParamService {
+  private activatedRoute = inject(ActivatedRoute);
+
+  get(queryParamName: string) {
+    const value$ = this.activatedRoute.queryParams.pipe(
+      map((allQueryParams) => allQueryParams[queryParamName])
+    );
+
+    return toSignal(value$);
+  }
+}
+```
+
+```ts
+@Component({
+  selector: "app-root",
+  standalone: true,
+  template: `
+    <h1>Signals Demo</h1>
+    <p>Full name: {{ fullName() }}</p>
+    <p>Age: {{ age() }}</p>
+
+    <button (click)="anonymise()">I want to be anonymous 👤</button>
+  `,
+})
+export class AppComponent {
+  private queryParamService = inject(QueryParamService);
+
+  firstName = this.queryParamService.get("firstName");
+  lastName = this.queryParamService.get("lastName");
+  age = this.queryParamService.get("age");
+
+  fullName = computed(() => `${this.firstName() ?? ""} ${this.lastName() ?? ""}`);
+
+  anonymise() {
+    // TODO: in a minute :)
+  }
+}
+```
+
+This small refactor will produce the same behavior as before except for the anonymising feature, which we will implement more adequately as soon as we make our `get` method from `QueryParamService` return a writable signal.
```

Comment:
i have no idea about commas in english, but for me it looks better if there is a comma before except and not before which
but tbh i have no clue :D